google/cloud/internal/curl_impl.cc

```diff
@@ -707,19 +715,23 @@ Status CurlImpl::MakeRequest(CurlImpl::HttpMethod method,
   if (method == HttpMethod::kDelete || payload.empty())
     return MakeRequestImpl();
 
-  // TODO(#7955): add support for multi-Span POST
   if (method == HttpMethod::kPost) {
-    if (payload.size() == 1) {
-      status = handle_.SetOption(CURLOPT_POSTFIELDSIZE, payload[0].size());
-      if (!status.ok()) return OnTransferError(std::move(status));
-      status = handle_.SetOption(CURLOPT_POSTFIELDS, payload[0].data());
-      if (!status.ok()) return OnTransferError(std::move(status));
-      return MakeRequestImpl();
-    }
-    return Status{
-        StatusCode::kInvalidArgument,
-        absl::StrCat(method, " with more than one Span not supported"),
-        {}};
+    WriteVector writev{std::move(payload)};
+    status = handle_.SetOption(CURLOPT_POSTFIELDSIZE, nullptr);
```

Comment:
Do you need to use `POSTFIELDSIZE` if you are going to use `POSTFIELDSIZE_LARGE`?